### PR TITLE
Fix eth bridge pause and validator set issue

### PR DESCRIPTION
**Pull request by JasonTulp**
This PR addresses two tickets:
- https://futureverse.atlassian.net/jira/software/c/projects/TRN/boards/39?selectedIssue=TRN-315
addressed by changing bridge paused type from a bool to a struct that contains two bools based on the pause origin. i.e. manual pause or authorities change pause.

- https://futureverse.atlassian.net/jira/software/c/projects/TRN/boards/39?selectedIssue=TRN-314
addressed by reassigning the validator_set_id when the delayed proof is sent, right before the event is thrown

### CODE REVIEW

**Comment by zees-dev** [comment]
rebase required.

File: pallet/ethy/src/lib.rs [hunk 1]
```diff
@@ -570,7 +570,7 @@ pub mod pallet {
 
 			// 3) Try process delayed proofs
 			consumed_weight += DbWeight::get().reads(2u64);
-			if PendingEventProofs::<T>::iter().next().is_some() && !BridgePaused::<T>::get() {
+			if !Self::bridge_paused() && PendingEventProofs::<T>::iter().next().is_some() {
```
Comment (zees-dev) [inline pallet/ethy/src/lib.rs:573]: nit: to remove level of nesting

```suggestion
			if Self::bridge_paused() || PendingEventProofs::<T>::iter().next().is_none() {
                return consumed_weight;
            }
            ...
```

File: pallet/ethy/src/lib.rs [hunk 2]
```diff
@@ -218,7 +218,7 @@ pub mod pallet {
 
 	/// Whether the bridge is paused (e.g. during validator transitions or by governance)
 	#[pallet::storage]
-	pub type BridgePaused<T> = StorageValue<_, bool, ValueQuery>;
+	pub type BridgePaused<T> = StorageValue<_, BridgePauseStatus, ValueQuery>;
```
Comment (surangap) [inline pallet/ethy/src/lib.rs:221]: Might need migrations since the data type is changed now

**Review by surangap — APPROVED** [review]
Lgtm, added a comment for migrations

File: pallet/ethy/src/lib.rs (hunk 2 above)
Comment (zees-dev) [reply pallet/ethy/src/lib.rs:221]: Was going to say the same.

**Review by zees-dev — COMMENTED** [review]
Needs migration I believe; otherwise looks good 👍

File: pallet/ethy/src/lib.rs (hunk 1 above)
Comment (JasonTulp) [reply pallet/ethy/src/lib.rs:573]: Leaving as is because this whole piece of code is refactored in #831

**Review by zees-dev — APPROVED** [review]
LGTM 👍

**Pull request merged**